### PR TITLE
SplitMetadataContributor.java: new class to split metadata values based on a regex

### LINKED ISSUE #7969: Bibtex import does not parse multiple authors into different fields
**Description**
In the DSpace 7 demo area I wanted to import a Bibtex file from the MyDSpace area. However, the authors were not parsed separately, but as one big block.

**Reproduce**
1. Get  a bibtex file with multiple authors
2. Go to MyDSpace
3. Select the bibtex file to import
4. Check the author field on the new item submission

**Expected behavior**
Multiple authors to be parsed into different fields

**Example Bibtex**
@article{Moreno-ParedesSantiago2021Fpae,
author = {Moreno-Paredes, Santiago and Rodriguez-Alcala, Laura and Martin-Lagos Martinez, Juan and Muller Locatelli, Nicolas and Vazquez Lopez, Cristina and Vargas Fernandez, Jose Luis and Cabrera Peea, Alvaro},
address = {England},
copyright = {COPYRIGHT 2021 BioMed Central Ltd.},
issn = {1471-2377},
journal = {BMC neurology},
keywords = {Adult ; Cardiovascular agents ; Case studies ; Complications and side effects ; Embolism - chemically induced ; Embolization ; Embolization, Therapeutic - adverse effects ; Embolization, Therapeutic - methods ; Endovascular Procedures - adverse effects ; Endovascular Procedures - methods ; Ethylene-vinyl alcohol copolymer ; Facial Palsy ; Facial Paralysis - etiology ; Female ; Head and Neck Neoplasms - therapy ; Humans ; Magnetic Resonance Imaging ; Neck paraganglioma ; Paraganglioma - therapy ; Paralysis, Facial ; Polyvinyls - adverse effects ; Squid 12 ; Therapeutic embolization},
language = {eng},
number = {1},
pages = {45-4},
publisher = {BioMed Central Ltd},
title = {Facial palsy after embolization with Squid® 12},
volume = {21},
year = {2021},
}

### PR DESCRIPTION
**Pull request by philipprumpf**
Needed for BibTeX's convention of separating author names by " and ". Code by @johannastaudinger (@uniba-ub).

## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #7969

## Description
Based on EnhancedSimpleMetadataContributor and the MappedMetadataContributor.java file in https://github.com/4Science/DSpace.

Configuration example (bibtex-integration.xml).

```
    <bean id="bibtexAuthorsContrib" class="org.dspace.importer.external.metadatamapping.contributor.SplitMetadataContributor">
        <constructor-arg name="regex" value="[\r\n ]and[\r\n ]" />
        <constructor-arg name="innerContributor" ref="bibtexAuthorContrib" />
    </bean>

    <bean id="bibtexAuthorContrib" class="org.dspace.importer.external.metadatamapping.contributor.SimpleMetadataContributor">
        <property name="field" ref="dcAuthors"/>
        <property name="key" value="author" />
    </bean>
```

## Instructions for Reviewers

List of changes in this PR:
* create new class

Tested on DSpace-CRIS only so far. No test cases yet.

## Checklist

- n/a My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [ ] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [ ] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [ ] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- n/a If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- n/a If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by tdonohue** [comment]
@philipprumpf : This PR looks to have checkstyle violations.  Could you correct them?  You can run `mvn checkstyle:check` locally to see them on your machine.

**Comment by philipprumpf** [comment]
Sorry about that! I'll make sure it won't happen again.

**Comment by tdonohue** [comment]
@philipprumpf : Could you rebase this PR against the latest `main`?  It now has some minor merge conflicts after I merged a few of your other BibTeX related PRs.  Once it's rebased, I can give this one a review/test too.  Thanks!

**Comment by philipprumpf** [comment]
@tdonohue Thank you! I've rebased and run the tests locally, seems to work.

File: dspace-api/src/main/java/org/dspace/importer/external/metadatamapping/contributor/SplitMetadataContributor.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.dspace.importer.external.metadatamapping.MetadataFieldMapping;
+import org.dspace.importer.external.metadatamapping.MetadatumDTO;
+
+public class SplitMetadataContributor<T> implements MetadataContributor<T> {
+
+    private final MetadataContributor<T> innerContributor;
+    private String regex;
+
+    public SplitMetadataContributor(MetadataContributor<T> innerContributor, String regex) {
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/importer/external/metadatamapping/contributor/SplitMetadataContributor.java:21]: Please also add JavaDocs to this constructor method, especially to describe the "regex" and how it is used (obviously this regex is used to split the values, but we should describe that in JavaDocs)

**Review by tdonohue — COMMENTED** [review]
@philipprumpf : Thanks!  The code here looks good overall & it works (I tested using a BibTex file with several authors separated by " and ").

However, I noticed that the new `SplitMetadataContributor` is missing JavaDocs.   Those are required for all new classes/methods.  If you could please add those, then this will be ready to merge.

**Comment by tdonohue** [comment]
@philipprumpf : Just a reminder that this PR is basically "approved" and ready to go...but it's waiting on JavaDocs to be added to document the new classes/methods.  See my [previous review](https://github.com/DSpace/DSpace/pull/8461#pullrequestreview-1093901591).  Once those are added, let me know & I can give this a final review & merger. Thanks!

**Review by tdonohue — APPROVED** [review]
👍 Thanks @philipprumpf !  This looks great now.  I gave it one final test today and found no issues.

**Pull request merged**